### DISCUSSION TITLE
Scrolling like in mantine.dev/theming/theme-object

### DISCUSSION THREAD

**Discussion by andrey-germanov** [mantinedev/mantine#3305] [Q&A] [answered]
Hi all.
I want to implement scrolling functionality like in https://mantine.dev/theming/theme-object/. 
I used the useIntersection hook. 
Wrapped the content in Paper. Everything seems to be fine, but the problem is that I need to specify a fixed height of the Paper. 
And my content is dynamic.

Tried passing ref to content wrapper and calculating contentRef.current?.scrollHeight and passing that to Paper. But it doesn't work.

Can you please tell me how to do it. Maybe I should not useIntersection.

**Comment by rtivital** [comment, marked as answer]
If you use ScrollArea, try replacing it with ScrollArea.Autosize – https://mantine.dev/core/scroll-area/#scrollareaautosize
You can also browse documentation source code here – https://github.com/mantinedev/mantine/tree/master/docs

**Reply by andrey-germanov** [reply]
Please tell me how scrolling is done here https://mantine.dev/theming/theme-object/
What was used there?

**Reply by rtivital** [reply]
You can find table of content code here – https://github.com/mantinedev/mantine/blob/modal-base/docs/src/components/MdxPage/TableOfContents/TableOfContents.tsx#L7
